sklearn/impute/_knn.py

```diff
@@ -0,0 +1,264 @@
+import numpy as np
+
+from ..base import BaseEstimator, TransformerMixin
+from ..utils.validation import FLOAT_DTYPES
+from ..metrics import pairwise_distances
+from ..metrics.pairwise import _NAN_METRICS
+from ..neighbors.base import _get_weights
+from ..neighbors.base import _check_weights
+from ..utils import check_array
+from ..utils import is_scalar_nan
+from ..utils.mask import _get_missing_mask
+from ..utils.validation import check_is_fitted
+
+
+class KNNImputer(TransformerMixin, BaseEstimator):
+    """Imputation for completing missing values using k-Nearest Neighbors.
+
+    Each sample's missing values are imputed using values from ``n_neighbors``
+    nearest neighbors found in the training set. Each missing feature is then
+    imputed as the average, either weighted or unweighted, of these neighbors.
+    If a sample has more than one feature missing, then the neighbors for that
+    sample can be different depending on the particular feature being imputed.
+    When the number of available neighbors is less than ``n_neighbors``, the
+    training set average for that feature is used during imputation. When a
+    sample has more than a ``feature_max_missing`` fraction of its features
```

Comment:
``feature_max_missing`` doesn't exist any more.

Reply:
If it's less than ``k_neighbors`` we're using the mean, right?

Reply:
Yes, I hope the sentence from the docstring is clear enough about this:

> When the number of available neighbors is less than ``n_neighbors``, the
    training set average for that feature is used during imputation.

Reply:
yeah that's fine.